### ISSUE TITLE
insert comma when creating newline in JSON

### ISSUE THREAD

**Issue by kurtbuilds** [zed-industries/zed#18509]
### Check for existing issues

- [X] Completed

### Describe the feature

In IntelliJ, the editor understands your code edit actions and the local context. As one example, using vim mode, press "o" (for insert new line) on the last line of a JSON object will insert the newline AND also insert a comma preceding that line.

It makes it much faster and easier to edit JSON, and it'd be huge benefit for Zed to do this as well.

**Comment by kurtbuilds** [comment]
Another issue is that Zed currently inserts `null` as the value when inserting new keys, when I would expect it to create a nested object.

Example when editing `settings.json`:

<img width="322" alt="image" src="https://github.com/user-attachments/assets/13651075-f83c-458e-b8e7-9a19843e9b1e">

<img width="320" alt="image" src="https://github.com/user-attachments/assets/3a897d14-e4d9-4784-a091-f8f6825e5d04">

Instead of an object, Zed creates a `null` value, which I then have to select, delete, create braces, and hit newline, to create the new object.

This might be a separate issue, as it depends on Zed's understanding of the jsonschema of the current document.

**Comment by notpeter** [comment]
Yes, it would be nice if Zed would automatically introduce a comma when appending an additional key to the end of an object.  Interestingly we already support removing an excess trailing comma automatically (prettier will remove it if `format_on_save` is on) and it always felt asymmetrical that Zed would remove an extra comma on save, but never add one back.

Anyways JetBrains "Smart Keys" does this by default, it'd be nifty if Zed could too. 
<img width="898" alt="Screenshot 2024-10-13 at 10 55 19" src="https://github.com/user-attachments/assets/e2e5758c-4104-4f82-a78b-eabe9fe43e81">

**Comment by nesl247** [comment]
This is still needed.

**Comment by ian-h-chamberlain** [comment]
It's not a complete fix (won't work for plain JSON), but for use with JSONC that supports trailing commas, and #43854 now merged, it's possible to automatically add trailing commas, which mitigates this issue. Configuration example:

```jsonc
{
  "prettier": {
    "overrides": [
      {
        "files": ["**/Zed/*.json", "**/zed/*.json", "*.jsonc"],
        "options": {
          "parser": "jsonc",
        },
      },
    ],
  },
}
```

With this, the builtin `prettier` integration will add trailing commas, making it a lot more ergonomic to add new items to these files.
